src/prefect/server/models/deployments.py

```diff
@@ -107,6 +107,11 @@ async def create_deployment(
 
     requested_concurrency_limit = insert_values.pop("concurrency_limit", "unset")
 
+    if requested_concurrency_limit != "unset" and deployment.concurrency_limit_id:
+        raise ValueError(
+            "Cannot provide both concurrency_limit and concurrency_limit_id"
+        )
```

Comment:
This validation should probably live on the `DeploymentCreate` model as a [model validator](https://docs.pydantic.dev/latest/concepts/validators/#model-validators).